bpf/http_sock.h

```diff
@@ -337,6 +338,9 @@ static __always_inline void process_http_response(http_info_t *info, unsigned ch
     info->status += (buf[RESPONSE_STATUS_POS]     - '0') * 100;
     info->status += (buf[RESPONSE_STATUS_POS + 1] - '0') * 10;
     info->status += (buf[RESPONSE_STATUS_POS + 2] - '0');
+    if (info->status > MAX_HTTP_STATUS) { // we read something invalid
```

Comment:
Defensive measure to at reset the status if we parsed something incorrect, allowing us to potentially find the correct response (or at least set 0).